src/gluonts/model/estimator.py

```diff
@@ -44,6 +44,14 @@ class Estimator:
 
     prediction_length: int
     freq: str
+    lead_time: int
+
+    def __init__(self, lead_time: int = 0, **kwargs) -> None:
+        # TODO validation of prediction_length and freq could also
+        # TODO be bubbled-up here from subclasses classes
+        assert lead_time >= 0, "The value of `lead_time` should be >= 0"
```

Comment:
I think having asserts like this can also be done for `prediction_length` and `freq` where through inheritance it avoids all the redundant checks for each estimator implementation.

Reply:
Yes, I agree